middleware/reverse/setup.go

```diff
@@ -0,0 +1,107 @@
+package reverse
+
+import (
+	"net"
+	"sort"
+	"strings"
+	"strconv"
+
+	"github.com/mholt/caddy"
+	"github.com/miekg/coredns/core/dnsserver"
+	"github.com/miekg/coredns/middleware"
+)
+
+func init() {
+	caddy.RegisterPlugin("reverse", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setupReverse,
+	})
+}
+
+func setupReverse(c *caddy.Controller) error {
+	networks, err := reverseParse(c)
+	if err != nil {
+		return err
+	}
+
+	dnsserver.GetConfig(c).AddMiddleware(func(next middleware.Handler) middleware.Handler {
+		return Reverse{Next: next, Networks:networks}
+	})
+
+	return nil
+}
+
+func reverseParse(c *caddy.Controller) (Networks, error) {
+	var err error
+	config := dnsserver.GetConfig(c)
+	networks := Networks{}
+
+	for c.Next() {
+		if c.Val() == "reverse" {
+
+			if !c.NextArg() || c.Val() == "{" {
+				return nil, middleware.Error("reverse", c.ArgErr())
```

Comment:
Just return it as an `error` and call `middleware.Error` once in the setup function.